### PR TITLE
Deploying Ceph Cluster with cephadm

### PR DESCRIPTION
**Pull request by tbazant**
This update introduces a new deployment method for SES7

### CODE REVIEW

File: xml/deploy_cephadm.xml [hunk 1]
```diff
@@ -5,22 +5,107 @@
...
+  </listitem>
+  <listitem>
+   <para>
+    Deploy the &salt; infrastructure over all cluster nodes so that you can
+    orchestrate them effectively.
+   </para>
+  </listitem>
+  <listitem>
+   <para>
+    Configure basic properties of the cluster and deploy it.
```
[the remark this answers is not in the document]
Comment (lproven) [reply xml/deploy_cephadm.xml:35]: Would "and bootstrap" or "and orchestrate" it be better than "and deploy it"?

File: xml/deploy_cephadm.xml [hunk 2]
```diff
@@ -5,22 +5,107 @@
...
+ </orderedlist>
  <sect1 xml:id="deploy-os">
-  <title>Install &cephos;</title>
+  <title>Install and Configure SLES</title>
 
-  <para></para>
+  <procedure>
+   <step>
+    <para>
+     Install and register &cephos; on each cluster node. Include the following
```
Comment (lproven) [inline xml/deploy_cephadm.xml:45]: I think we should also specify that they should do a "minimal" installation  -- no X etc. -- and that using JeOS is an option. What do you think?

File: xml/deploy_cephadm.xml [hunk 3]
```diff
@@ -5,22 +5,107 @@
...
+      </para>
+     </listitem>
+     <listitem>
+      <para>
+       Containers Module
+      </para>
+     </listitem>
+    </itemizedlist>
+    <para>
+     Find more details on how to install SLES in
```
Comment (lproven) [inline xml/deploy_cephadm.xml:66]:
```suggestion
     Find more details on how to install &sles; in
```

File: xml/deploy_cephadm.xml [hunk 4]
```diff
@@ -5,22 +5,107 @@
...
+     <link xlink:href="https://documentation.suse.com/sles/15-SP22/html/SLES-all/cha-install.html"/>.
+    </para>
+   </step>
+   <step>
+    <para>
+     Install the <emphasis>&productname; &productnumber;</emphasis> extension
+     on each cluster node.
+    </para>
+    <tip>
+     <title>Install SES Together with SLES</title>
```
Comment (lproven) [inline xml/deploy_cephadm.xml:76]:
```suggestion
     <title>Install SES Together with &sles;</title>
```

Comment (lproven) [reply xml/deploy_cephadm.xml:76]: Shouldn't we have entities for SES and SUMA?

File: xml/deploy_cephadm.xml [hunk 5]
```diff
@@ -5,22 +5,107 @@
...
+   </step>
+  </procedure>
  </sect1>
  <sect1 xml:id="deploy-salt">
   <title>Deploy &salt;</title>
 
+  <para>
+   &productname; uses &salt; as a cluster orchestrator. &salt; helps you
+   configure and run commands on multiple cluster nodes simultaneously from one
+   dedicated host called <emphasis>&smaster;</emphasis>. Before deploying
```
Comment (lproven) [inline xml/deploy_cephadm.xml:105]:
```suggestion
   dedicated host called the <emphasis>&smaster;</emphasis>. Before deploying
```

File: xml/deploy_cephadm.xml [hunk 6]
```diff
@@ -5,22 +5,107 @@
...
+   &salt;, consider the following valid points:
```
Comment (lproven) [inline xml/deploy_cephadm.xml:106]:
```suggestion
   &salt;, consider the following important points:
```

File: xml/deploy_cephadm.xml [hunk 7]
```diff
@@ -51,50 +136,508 @@
...
+    <para>
+     Check that the <systemitem>salt-master</systemitem> service is enabled and
+     started, and enable and start it if needed:
+    </para>
+<screen>&prompt.smaster;systemctl enable salt-master.service
+&prompt.smaster;systemctl start salt-master.service</screen>
+   </step>
+   <step>
+    <para>
+     If you intend to use firewall, verify that the &smaster; node has ports
```
Comment (lproven) [inline xml/deploy_cephadm.xml:160]:
```suggestion
     If you intend to use the firewall, verify that the &smaster; node has ports
```

File: xml/deploy_cephadm.xml [hunk 8]
```diff
@@ -51,50 +136,508 @@
...
+    </para>
+   </step>
+   <step>
+    <para>
+     Install the package <literal>salt-minion</literal> on all minion nodes.
+    </para>
+<screen>&prompt.sminion;zypper in salt-minion</screen>
+    <para>
+     Make sure that the <emphasis>fully qualified domain name</emphasis> of
+     each node can be resolved to the public network IP address by all other
```
Comment (lproven) [inline xml/deploy_cephadm.xml:173]:
```suggestion
     each node can be resolved to an IP address on the public network by all the other
```

Comment (lproven) [reply xml/deploy_cephadm.xml:173]: If I understand this correctly -- the FQDN should be on the public network, yes? Not any other, e.g. an inter-node network?

File: xml/deploy_cephadm.xml [hunk 9]
```diff
@@ -51,50 +136,508 @@
...
+</screen>
   </sect2>
 
-  <sect2 xml:id="deploy-methods-orchestrator">
-   <title>&ceph; Orchestrator</title>
+  <sect2 xml:id="deploy-cephadm-configure">
+   <title>Configure Cluster Properties</title>
    <para>
-    EMPTY
+    Use the <command>ceph-salt config</command> command to configure basic
```
Comment (lproven) [inline xml/deploy_cephadm.xml:285]:
```suggestion
    Use the <command>ceph-salt config</command> command to configure the basic
```

File: xml/deploy_cephadm.xml [hunk 10]
```diff
@@ -51,50 +136,508 @@
...
+      <listitem>
+       <para>
+        To let the shell finish an absolute path, enter <keycap>/</keycap> and
+        hit the TAB key <keycap function="tab"></keycap> twice.
+       </para>
+      </listitem>
+     </itemizedlist>
+    </tip>
+    <tip>
+     <title>Navigating with Cursor Keys</title>
```
Comment (lproven) [inline xml/deploy_cephadm.xml:382]:
```suggestion
     <title>Navigating with the Cursor Keys</title>
```

File: xml/deploy_cephadm.xml [hunk 11]
```diff
@@ -51,50 +136,508 @@
...
+&prompt.smaster;ceph-salt status
+hosts:  0/5 deployed
+config: OK
+</screen>
+    </tip>
+   </sect3>
+   <sect3 xml:id="deploy-cephadm-configure-export">
+    <title>Export Cluster Configuration</title>
+    <para>
+     After you configured the basic cluster and its configuration is valid, it
```
Comment (lproven) [inline xml/deploy_cephadm.xml:599]:
```suggestion
     After you have configured the basic cluster and its configuration is valid, it
```

File: xml/leftovers.xml [hunk 12]
```diff
@@ -0,0 +1,89 @@
+<!-- ZAPPING OSD DISKS -->
+<step xml:id="deploy-wiping-disk">
+ <para>
+  Prior to deploying &productname; &productnumber;, manually zap all the
```
Comment (lproven) [inline xml/leftovers.xml:4]:
```suggestion
  Prior to deploying &productname; &productnumber;, manually zero all the
```

File: xml/leftovers.xml [hunk 13]
```diff
@@ -0,0 +1,89 @@
...
+   <para>
+    If the disk is managed by LVM, deactivate and delete the whole LVM
+    infrastructure. Refer to
+    <link xlink:href="https://documentation.suse.com/sles/15-SP2/html/SLES-all/cha-lvm.html"/>
+    for more details.
+   </para>
+  </step>
+  <step>
+   <para>
+    If the disk is part of MD RAID, deactivate the RAID. Refer to
```
Comment (lproven) [inline xml/leftovers.xml:29]:
```suggestion
    If the disk is part of an <literal>mdraid</literal>, deactivate the array. Refer to
```

File: xml/deploy_cephadm.xml (hunk 2 above)
Comment (tbazant) [reply xml/deploy_cephadm.xml:45]: No, JeOS never was a requirement, let's wait what SES team comes with

File: xml/deploy_cephadm.xml [hunk 14]
```diff
@@ -51,50 +136,508 @@
...
+    </para>
+    <tip>
+     <title>&smaster; and &adm; on the Same Host</title>
+     <para>
+      In a homogeneous environment where all or most nodes belong to SES, we
+      recommend having the &adm; on the same host as the &smaster;.
+     </para>
+     <para>
+      In a heterogeneous environment where one &salt; infrastructure hosts more
+      than one cluster, for example, &productname; together with &suma;, do
```
[the remark this answers is not in the document]
Comment (tbazant) [reply xml/deploy_cephadm.xml:443]: fixed

**Pull request merged**